amida-tech/recruitment-registry
# initial implementation of scale type

If this PR fixes a bug, you _must_ add test cases representative of the bug.

Please refer to [Tettra](https://app.tettra.co/teams/amida/pages/amida-pull-request-and-code-review-guide) for PR review guidelines.

#### What does this PR do?
This adds the "scale" question type, which is allows min and max limits on a number input. This was adapted from a similar feature added to survey service. See https://github.com/amida-tech/amida-survey-microservice/pull/38

#### Related JIRA tickets:
RR-756

#### How should this be manually tested?
Create/get/export/import scale questions and answer them.

1. Scale questions are specified using type scale and property meta: { scaleLimits: {min: , max: }}.
2. Min and max are optional but at least one of them should be present.
3. When both present max needs to be greater or equal to min.
4. Answers to scale questions are specified using numberValue which accepts float.
5. Answers will be validated when posted according to min and max.

#### Background/Context

#### Screenshots (if appropriate):
